optimum/gptq/quantizer.py

```diff
@@ -0,0 +1,571 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import copy
+import json
+import os
+from logging import getLogger
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import torch
+from torch import nn
+from transformers.pytorch_utils import Conv1D
+
+from ..utils import is_accelerate_available, is_autogptq_available
+from .constants import GPTQ_CONFIG
+from .data import get_dataset, prepare_dataset
+from .utils import get_block_name, get_device, get_layers, get_module_by_name_prefix, get_preceding_modules, get_seqlen
+
+
+if is_accelerate_available():
+    from accelerate import Accelerator, load_checkpoint_and_dispatch
+
+if is_autogptq_available():
+    from auto_gptq.quantization import GPTQ
+    from auto_gptq.utils.import_utils import dynamically_import_QuantLinear
+
+logger = getLogger(__name__)
+
+
+class GPTQQuantizer(object):
+    r"""
+    A simple API for GTPQ Quantization
+    """
+
+    def __init__(
+        self,
+        bits: int,
+        group_size: int = 128,
+        damp_percent: float = 0.01,
+        desc_act: bool = True,
+        sym: bool = True,
+        true_sequential: bool = True,
+        pack_sequentially: bool = False,
+        use_cuda_fp16: bool = True,
+        model_seqlen: Optional[int] = None,
+        block_name_to_quantize: Optional[str] = None,
+        module_name_preceding_first_block: Optional[List[str]] = None,
+        *args,
+        **kwargs,
+    ):
+        """
+        Args:
+            bits (`int`):
+                The number of bits to quantize to, supported numbers are (2, 3, 4, 8).
+            group_size (int, *optional*, defaults to -1):
+                The group size to use for quantization. Recommended value is 128 and -1 uses full row.
+            damp_percent (`float`, *optional*, defaults to `0.01`):
+                The percent of the average Hessian diagonal to use for dampening, recommended value is 0.01.
+            desc_act (`bool`, *optional*, defaults to `True`):
+                Whether to quantize columns in order of decreasing activation size.
+                Setting it to False can significantly speed up inference but the perplexity may become slightly worse.
+                Also known as act-order.
+            sym (`bool`, *optional*, defaults to `True`):
+                Whether to use symetric quantization.
+            true_sequential (`bool`, *optional*, defaults to `True`):
+                Whether to performing sequential quantization even within a single Transformer block.
```

Comment:
Can you explain in the docstring what sequential is? It is not mentioned in GPTQ paper and the original repo is not explicit about it.
```suggestion
                Whether to perform sequential quantization even within a single Transformer block.
```

Reply:
Yes, I will do that. If you want more information you can check this [link](https://github.com/IST-DASLab/gptq#new-features) from the original repo of gtpq. They say that it improves the performance of the model.